### DISCUSSION TITLE
Menus within Modals (Dialogs)

### DISCUSSION THREAD

**Discussion by Southclaws** [cschroeter/park-ui#428] [General]
I'm trying to get a Menu component working inside a Dialog, the first roadblock I hit was that the default z-index for Menu seems to be "dropdown" (1000) which is lower than "modal" (1400) so I fixed that by changing the Menu recipe to use "popover" (1500) for the z-index, after this fix the Menu is now correctly rendering on top of Dialog components instead of behind.

But now, whenever I mouse-over the items in the menu, or click anywhere in the `<Menu.Content>`, it closes. I'm not sure what's going on or even where, it could be any one of: Zag, Ark, Panda or Park (separation of concerns can be exhausting!)

Any ideas? I've just recently upgraded Ark and Panda to their latest versions (which in turn upgrades Zag too) so I'm unsure what's causing this.

**Comment by dannylin108** [comment]
I found similar bug for touch devices

https://github.com/chakra-ui/ark/issues/2816

**Comment by segunadebayo** [comment]
This is more of a layering or DOM issue. It sounds like the menu component is portalled as well.

I recommend portalled just the dialog and leave the menu within the dialog (non-portalled). That way, you don't have to fix z-index related issues.
